addon/components/polaris-range-slider.js

```diff
@@ -0,0 +1,289 @@
+   * @public
+   */
+  onBlur() {},
+
+  /**
+   * Class names for the range input wrapper div
+   *
+   * @property {rangeInputWrapperClassNames}
+   * @type {String}
+   * @private
+   */
+  rangeInputWrapperClassNames: computed('error', 'disabled', function() {
+    let { error, disabled } = this.getProperties('error', 'disabled');
+    let classNames = ['Polaris-RangeSlider'];
+
+    if (error) {
+      classNames.push('Polaris-RangeSlider--error');
+    }
+
+    if (disabled) {
+      classNames.push('Polaris-RangeSlider--disabled');
+    }
+
+    return classNames.join(' ');
+  }).readOnly(),
+
+  /**
+   * Style for the range input wrapper div
+   *
+   * @property {rangeInputWrapperStyle}
+   * @type {String}
+   * @private
+   */
+  rangeInputWrapperStyle: computed('min', 'max', 'value', 'sliderProgress', function() {
+    let { min, max, value: current, sliderProgress } = this.getProperties('min', 'max', 'value', 'sliderProgress');
+    let styleProps = assign({ min, max, current }, {
+      current,
+      progress: `${ sliderProgress }%`,
+      outputFactor: invertNumber((sliderProgress - 50) / 100),
+    });
+
+    let styleString = Object.keys(styleProps).reduce((styleString, propName) => {
+      return `${ styleString } --Polaris-RangeSlider-${ dasherize(propName) }:${ styleProps[propName] };`;
+    }, '');
+
+    return htmlSafe(styleString.trim());
+  }).readOnly(),
+
+  /**
+   * Slider progress percentage
+   *
+   * @property {sliderProgress}
+   * @type {Number}
+   * @private
+   */
+  sliderProgress: computed('min', 'max', 'value', function() {
+    let { min, max, value } = this.getProperties('min', 'max', 'value');
+    return ((value - min) * 100) / (max - min);
+  }).readOnly(),
+
+  /**
+   * Stringified boolean flag indicating whether an error is present
+   *
+   * @property {hasError}
+   * @type {String}
+   * @private
+   */
+  hasError: computed('error', function() {
+    return Boolean(this.get('error')).toString();
+  }).readOnly(),
+
+  /**
+   * Accessibility
+   *
+   * @property {ariaDescribedBy}
+   * @type {Boolean}
+   * @private
+   */
+  ariaDescribedBy: computed('error', 'helpText', function() {
+    let { error, helpText, id } = this.getProperties('error', 'helpText', 'id');
+    let describedBy = [];
+
+    if (error) {
+      describedBy.push(errorId(id));
+    }
+
+    if (helpText) {
+      describedBy.push(helpTextId(id));
+    }
+
+    return describedBy.length ? describedBy.join(' ') : undefined;
+  }).readOnly(),
+
+  /**
+   * Boolean flag indicating whether the output value should be displayed
+   *
+   * @property {shouldShowOutput}
+   * @type {Boolean}
+   * @private
+   */
+  shouldShowOutput: computed('disabled', 'output', function() {
+    let { disabled, output } = this.getProperties('disabled', 'output');
+    return !disabled && output;
+  }).readOnly(),
+
+  actions: {
+    handleChange(event) {
+      let onChange = this.get('onChange');
+      if (isNone(onChange)) {
```

Comment:
since we make this a no-op for us, we can probably remove this?

Reply:
Can do, was just following the React implementation which has `onChange` as a required argument but still makes this check 🤷‍♂️

Reply:
yeah, fine with me to leave it too....nothing bad here